### DISCUSSION TITLE
How to enforce private pages for private repos, but allow public ones too, in GH Enterprise

### DISCUSSION THREAD

**Discussion by dayalstrub-cma** [community/community#23438] [Enterprise]
<p>Hi all, and sorry if I haven’t found this in the docs or previous posts.</p>
<p>Is there a way in GitHub Enterprise to ensure/restrict private repositories to have private GitHub Pages, while still allowing public repos to have public Pages?</p>
<p>I can easily either set all Pages to be private, but if I select both public and private on the Enterprise settings, then all members seem to have the option to publish both public and private Pages on any of the Enterprise’s repositories.</p>
<p>Thanks!</p>

**Comment by yoannchaudet** [comment]
<p>Hello <a class="mention" href="/dayalstrub-cma">@dayalstrub-cma</a>,</p>
<p>No unfortunately this is not a feature that exists today. The public/private flag is controlling the whole appliance and as such apply to all Pages sites instead of select ones.</p>

**Comment by dayalstrub-cma** [comment]
<p>Thanks <a class="mention" href="/yoannchaudet">@yoannchaudet</a></p>
<p>I was hoping to get the best of both worlds, but still have in place some security without having to restrict to who can create a new repo within the Enterprise, and therefore be repo admin. Probably an edge use case.</p>
